t_sparse_list.py

```diff
@@ -300,5 +300,18 @@ def test_remove_default_value_does_nothing(self):
         sl.remove(None)
         self.assertEquals([None, None, None, None], sl)
 
+    def test_set_slice(self):
+        sl = sparse_list.SparseList(4, None)
+        # Only the first two should be assigned.
+        sl[0:2] = [1, 2, 3]
+        self.assertEquals([1, 2, None, None], sl)
+        # Like a normal list, when we assign past the end, it appends.
+        sl[100:] = [4, 5]
+        self.assertEquals([1, 2, None, None, 4, 5], sl)
+        self.assertEquals(len(sl), 6)
+        # Now assign from the middle, past the end.
+        sl[5:] = [6, 7, 8]
+        self.assertEquals([1, 2, None, None, 4, 6, 7, 8], sl)
```

Comment:
I find this less surprising, but this test relies on previous state.

Let's break this this into separate tests.